src/main/java/com/redhat/management/approval/EmailBody.java

```diff
@@ -64,10 +67,11 @@ public static String getUrlContent(URL url) throws Exception {
         return content.toString();
     }
 
-    public HashMap<String, String> getRequestParameters() {
-        Stage currentStage = EmailDispatcher.getCurrentStage(group, stages);
-        HashMap<String, Object> request_content = request.getContent();
-        HashMap<String, String> values = new HashMap<String, String>();
+    public Map<String, String> getRequestParameters() {
+        Stage currentStage = ApprovalApiHelper.findStageByGroup(group, stages);
+        Map<String, Object> request_content = request.getContent();
+
+        Map<String, String> values = new HashMap<String, String>();
```

Comment:
extract new line 80 - 110 into a separate method. This method is only to put map key/value pairs.

In addition, can `getUrlContent()` be a non static private method?